app/blueprints/advance_payments.py

```
@@ -0,0 +1,139 @@
+from sqlalchemy.exc import IntegrityError, InvalidRequestError, DataError
+from flask import Blueprint
+from app.tools.useful_functions import id_from_str_to_int, amount_parser
+from app.database.tables_declaration import *
+
+advance_payments_bp = Blueprint('advance_payments', __name__)
+
+
+@advance_payments_bp.route('/settlements/<settlement_id>/advance_payments', methods=['GET'])
+@Users.is_logged_in
```

Why it is a class method?
It is ok to have such function, but:
 1 - it should not be a class method without a reason
 2 - better to throw an exception (because bad request is a exceptional situation) instead of returning "response"
      now only "views" can be wrapped, but not any custom function
      read docs https://werkzeug.palletsprojects.com/en/2.0.x/exceptions/#module-werkzeug.exceptions